doc/source/whatsnew/v0.25.0.rst

```diff
@@ -252,6 +252,7 @@ Other API Changes
 - The ``arg`` argument in :meth:`pandas.core.groupby.DataFrameGroupBy.agg` has been renamed to ``func`` (:issue:`26089`)
 - The ``arg`` argument in :meth:`pandas.core.window._Window.aggregate` has been renamed to ``func`` (:issue:`26372`)
 - Most Pandas classes had a ``__bytes__`` method, which was used for getting a python2-style bytestring representation of the object. This method has been removed as a part of dropping Python2 (:issue:`26447`)
+- Bug in :func:`_ensure_valid_index` empty list like object should have an empty Index (:issue:`17101`)
```

Comment:
This can go under bug fixes, not API changes. In the "Indexing" subsection.

The release notes are for users, and `_ensure_valid_index` isn't a public function. Instead, write these as the behavior that caused the issue. Something like

    Bug when adding a new column to a length-zero ``DataFrame`` converting the index to a ``RangeIndex`` (:issue:`17101`).

Reply:
Done.